pulp_container/app/registry_api.py

```diff
@@ -392,7 +392,16 @@ def get_dr_push(self, request, path, create=False):
 
     def create_dr(self, path, request):
         with transaction.atomic():
-            repository = serializers.ContainerPushRepositorySerializer.get_or_create({"name": path})
+            try:
+                repository = serializers.ContainerPushRepositorySerializer.get_or_create(
+                    {"name": path}
+                )
+            except models.ContainerPushRepository.DoesNotExist:
+                raise RepositoryInvalid(
+                    name=path,
+                    message='Cannot create "' + path + '" push repository, another repository with '
+                    "the same name already exists.",
```

Comment:
Too much information. Let's stick to raising just `RepositoryInvalid(name=path, message="Repository is read-only.")`.